torchaudio/backend/_soundfile_backend.py

```diff
@@ -177,7 +182,11 @@ def save(
             "The argument is silently ignored."
         )
 
-    ext = str(filepath).split(".")[-1].lower()
+    try:
```

Comment:
For now we could simply require the user to pass a format for a file-like object and then ask for feedback and see whether there's interest in this soundfile-like behavior that autodetect the destination format in some cases.